### ISSUE TITLE
`full` should take an integer size

### ISSUE THREAD

**Issue by pmeier** [pytorch/pytorch#70906]
The [array API specification stipulates](https://data-apis.org/array-api/latest/API_specification/creation_functions.html?highlight=full#full-shape-fill-value-dtype-none-device-none) that the `size` parameter can also be an `int`, which should be treated the same as a sequence of `int`'s with a single element.

**Comment by mruberry** [comment]
Is the general idea that shapes (let's prefer "shape" to "size") can be specified as either a sequence of integers or single integer? If so, would PyTorch need changes beyond `full`? 

PyTorch also has (kinda weird) behavior where we allow sequences of integers to be specified like varargs when calling operations as methods. How would that interact with this proposal?

**Comment by pmeier** [comment]
> Is the general idea that shapes (let's prefer "shape" to "size") can be specified as either a sequence of integers or single integer?

IIUC, yes. But let's ask @rgommers to be sure.

> If so, would PyTorch need changes beyond full?

The other creation ops that currently specified by the array API that have a shape parameter are `empty`, `ones`, `zeros`. They already support integer shapes due to their `*shape` behavior.

> PyTorch also has (kinda weird) behavior where we allow sequences of integers to be specified like varargs when calling operations as methods. How would that interact with this proposal?

The functionality is roughly

```python

def foo(*shapes, ...):
        if len(shapes) == 1 and isinstance(shapes[0], tuple):
            shape = shapes[0]
        else:
            shape = shapes
```

Thus, `foo(1)` is internally turned into `(1,)`.

`torch.full` does not have the `*shapes` behavior since the user needs to be able pass the `fill_value` positionally after the shape. Thus, we need to special case it or in general all tensor creation functions that take more than just the shape positionally.

**Comment by rgommers** [comment]
> > Is the general idea that shapes (let's prefer "shape" to "size") can be specified as either a sequence of integers or single integer?
> 
> IIUC, yes. But let's ask @rgommers to be sure.

Yes indeed, it's simple user convenience: `full(3, fill_value=99)` is easier to write than `full((3,), fill_value=99)`
